### PR TITLE
enhancement(cli): Fix API client subscription drop

### LINKED ISSUE #9206: Dropping an API client should clean up its subscriptions
GraphQL subscriptions via the Vector API client are currently long-running and cannot be terminated, breaking compatibility with `futures::executor::block_on` that will never return since a subscription continues to run, even if the caller has returned.

A subscription client should drop any active WebSocket connections when the client itself falls out of scope.

### PR DESCRIPTION
**Pull request by leebenson**
Follow-up to #9139. Closes #9206.

Adds graceful shutdown to API client subscriptions on drop by returning from spawned loops. This prevents `top` from having to hard shut down the process on exit, instead propagating `exitcode::OK` to the top-level.

Signed-off-by: Lee Benson <lee@leebenson.com>

### CODE REVIEW

File: src/top/cmd.rs [hunk 1]
```diff
@@ -64,11 +64,7 @@ pub async fn cmd(opts: &super::Opts) -> exitcode::ExitCode {
 
     // Initialize the dashboard
     match init_dashboard(url.as_str(), opts, sender).await {
-        // Even though we could return an `exitcode::OK` here, the upstream block_on won't
-        // return due to lingering subscriptions, which manifests as a hung cursor in the
-        // terminal. To work around this for now, exit immediately.
-        // TODO: https://github.com/vectordotdev/vector/issues/9206
-        Ok(_) => std::process::exit(0),
+        Ok(_) => exitcode::OK,
```
Comment (jszwedko) [inline src/top/cmd.rs:67]: Happy to see this go away 😄

**Review by jszwedko — APPROVED** [review]
Nice clean-up, thanks!

File: src/top/cmd.rs (hunk 1 above)
Comment (leebenson) [reply src/top/cmd.rs:67]: RIP short-lived comment. 💀

Comment (jszwedko) [reply src/top/cmd.rs:67]: 😄 you never know which ones will stay around longer than expected.

**Pull request merged**